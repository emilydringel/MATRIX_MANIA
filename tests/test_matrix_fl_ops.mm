def int main(){
    matrix<float> m1 = [.1,.2,.3;.4,.5,.6];
    matrix<float> m2 = [.6,.5,.4;.3,.2,.1];
    printmf(m1+m2);
    printmf(m1-m2);
    printmf(m1*3);
    printmf(3*m1);
    printmf(m1*3.5);
    printmf(3.5*m1);
    matrix<float> m3 = [1.1,2.2;3.3,4.4;5.5,6.6];
    printmf(m1*m3);
    print(m1==m3);
    print(m1!=m3);
}