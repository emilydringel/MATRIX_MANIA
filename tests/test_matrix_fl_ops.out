0.700000 0.700000 0.700000
0.700000 0.700000 0.700000
-0.500000 -0.300000 -0.100000
0.100000 0.300000 0.500000
0.300000 0.600000 0.900000
1.200000 1.500000 1.800000
0.300000 0.600000 0.900000
1.200000 1.500000 1.800000
0.350000 0.700000 1.050000
1.400000 1.750000 2.100000
0.350000 0.700000 1.050000
1.400000 1.750000 2.100000
2.420000 3.080000
5.390000 7.040000
0
1